[Germaine Seah] iP

driip-ddrop opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” - David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke helps you remember the things you need to do. It is
* text-based
* **effortless** to learn
* ~fast~ extremely fast to use

To download Duke,
1.download [Duke](https://github.com/driip-ddrop/ip)
2.double-click it
3.add your task
4.let it manage your task for you :sunglasses:

And it is **FREE** :star_struck:

Features:
- [x] Managing tasks
- [x] Managing deadline
- [x] Managing events
- [ ] Reminders (coming soon)
#
If you are a Java Programmer, you can use it to practice Java too. Here’s the `main` method:

```Java
public class Duke {
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Commit:
Add Gradle support

driip-ddrop committed:
Level-1

driip-ddrop committed:
Level-2

driip-ddrop committed:
Level-3

driip-ddrop committed:
Level-4

driip-ddrop committed:
Level-5

driip-ddrop committed:
Level-6

driip-ddrop committed:
A-TextUiTesting

driip-ddrop committed:
Implement Level-7

driip-ddrop committed:
Implement Level-8

driip-ddrop committed:
Merge branch 'branch-Level-7'

driip-ddrop committed:
Merge branch 'branch-Level-8'

driip-ddrop committed:
Refactor code into Ui, Storage, Parser and TaskList classes

driip-ddrop committed:
Divide classes into packages

driip-ddrop committed:
Add JUnit tests

driip-ddrop committed:
Package the App as a JAR file

driip-ddrop committed:
Add JavaDoc comments to the code.

driip-ddrop committed:
Tweak the code to comply with a coding standard

driip-ddrop committed:
Implement Level-9: Add find feature

driip-ddrop committed:
Merge branch 'branch-A-JavaDoc'

driip-ddrop committed:
Merge branch 'branch-A-CodingStandard'

driip-ddrop committed:
Merge branch 'branch-Level-9'

driip-ddrop committed:
Edit the code to comply with coding standard

driip-ddrop committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

driip-ddrop committed:
Automate project builds using Gradle

driip-ddrop committed:
Use CheckStyle

lethiciars commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+     * @throws IOException if it is unable to read the file, file format is wrong
+     */
+    public ArrayList<Task> txtToArrayList() throws IOException {
+        ArrayList<Task> list = new ArrayList<>();
```
Perhaps naming the ArrayList tasklist would be more descriptive?

joenzkc replied:
tasks may be appropriate as well

lethiciars commented on src/main/java/duke/exception/IncorrectFormatException.java:
```diff
@@ -0,0 +1,8 @@
+package duke.exception;
+
+public class IncorrectFormatException extends DukeException{
```
I like how you created different subclasses of DukeException based on the type of user mistakes.

lethiciars commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+
+public class Deadline extends Task{
+
+    private LocalDateTime due;
```
Perhaps using a noun such as dueDate for the DateTime object name is clearer?

lethiciars commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+    /**
+     * Marks the task as completed
+     */
+    public void checkOffTask() {
```
I like how intuitive the method name is.

lethiciars reviewed (commented):
Overall clean and readable code, no major guideline violations. :-)

BananaTechs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+                    String keyword = message.substring(5);
+                    taskList.findTask(keyword);
+
+                }else if (message.startsWith("todo")) {
```
The spaces before these elses got deleted

BananaTechs commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return the String that has been formatted
+     */
+    public String displayTask() {
```
Maybe this could be renamed to toString()?

BananaTechs reviewed (commented):
I didn't really notice any significant coding standard violations! Really organised code, it's given me some ideas on how I can improve mine. Thanks!

joenzkc commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+
+public class Parser {
+
+    private TaskList taskList;
```
Would tasks be a better name? Since plural form should be used on names representing a collection of objects 🤔

joenzkc commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+        sc.close();
+    }
+
+    public void isValidEntry(String message, String type) throws DescriptionEmptyException {
```
Maybe a bit nitpicky, but perhaps the method names could be checkValidEntry, checkCorrectFormat, throwNotValid or something along those lines since methods should be verbs.

joenzkc commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,23 @@
+package duke.task;
+
+public class Todo extends Task{
```
ToDo might be more appropriate (PascalCase)

joenzkc reviewed (commented):
Overall, the code follows the coding standard quite well! Had to be really nit-picky to mention anything, keep up the good work! 👍

driip-ddrop committed:
Add a GUI to Duke

driip-ddrop committed:
Add a GUI to Duke

driip-ddrop committed:
Update visuals of GUI

driip-ddrop committed:
Implement A-Assertions

Use assert statements to document important assumptions that should hold
 at various points in the code.

driip-ddrop committed:
Implement Update Feature

To support a way to update existing tasks without having to delete it
first.

driip-ddrop committed:
Improve Code Quality

Minimize code duplication and removed dead code to improve code quality.

driip-ddrop committed:
Merge pull request #1 from driip-ddrop/branch-A-Assertions

Implement A-Assertions

driip-ddrop committed:
Merge branch 'master' into branch-A-CodeQuality

driip-ddrop committed:
Merge pull request #2 from driip-ddrop/branch-A-CodeQuality

Improve Code Quality

driip-ddrop committed:
Create gradle.yml

driip-ddrop committed:
Edit gradle.yml

driip-ddrop committed:
Edit gradle.yml

driip-ddrop committed:
Merge branch 'branch-A-CheckStyle'

driip-ddrop committed:
Delete .github/workflows directory

driip-ddrop committed:
Minor Tweaks in code

driip-ddrop committed:
Set theme jekyll-theme-minimal

driip-ddrop committed:
Merge branch 'branch-C-Update'

# Conflicts:
#	src/main/java/duke/task/TaskList.java

driip-ddrop committed:
Merge remote-tracking branch 'origin/master'

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Set theme jekyll-theme-cayman

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Update README.md

driip-ddrop committed:
Add JavaDoc to code

driip-ddrop committed:
Merge remote-tracking branch 'origin/master'

driip-ddrop committed:
Minor improvements to code

driip-ddrop committed:
Improve the GUI

driip-ddrop committed:
Improve the GUI

driip-ddrop committed:
Add representative screenshot

driip-ddrop committed:
Update README.md
